packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -264,14 +263,17 @@ export class ScopeHoistingPackager {
   }
 
   shouldBundleQueue(bundle: NamedBundle): boolean {
+    let referencingBundles = this.bundleGraph.getReferencingBundles(bundle);
+    let hasHtmlReference = referencingBundles.some(b => b.type === 'html');
```

Comment:
What would happen if we stopped using HTML as an entry point and instead used JS entry points?

Reply:
Yeah, worthwhile adding a test for it. But otherwise the premise seems fine

Reply:
Well it just wouldn't work lol. TBH so many other things would break if we did that this would be just one of many.

Reply:
I'm curious about this. We have hacks in place internally to help with what I think is the same problem y'all are trying to address with the async bundle runtime. The only rub is that we use JS entry points. Anecdotally I've felt like things work a little funky with JS entrypoints and I'm wondering if we should shift to using an html entrypoint. Any more context on why html entries are your preferred approach?